app/src/main/java/io/halkyon/services/BindApplicationService.java

```diff
@@ -29,6 +31,11 @@ public class BindApplicationService {
     public static final String USERNAME_KEY = "username";
     public static final String PASSWORD_KEY = "password";
 
+    public static final String VAULT_KV_PATH_KEY = "vault-path";
+
+    @Inject
+    VaultKVSecretEngine kvSecretEngine;
```

Comment:
If you hard code the dependency with Vault here, how will it be possible to switch to use another Secret Management system ?

Reply:
At the moment this is a poc and I was trying to use Vault. For using another management system we need to put in place an abstraction layer to hide it.